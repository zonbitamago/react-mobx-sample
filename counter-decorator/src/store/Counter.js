import { observable, computed } from "mobx";

class Counter {
  @observable counter = 0;
}

export default Counter;
