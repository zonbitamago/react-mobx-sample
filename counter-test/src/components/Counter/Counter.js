import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Counter.css";
import { observer } from "mobx-react";

@observer
class Counter extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="Counter">
        <p>Counter:{this.props.store.counter}</p>
        <button id="button" onClick={this.props.store.plus}>
          Add
        </button>
        <button id="minus" onClick={this.props.store.minus}>
          minus
        </button>
      </div>
    );
  }
}

Counter.propTypes = {};

Counter.defaultProps = {};

export default Counter;
