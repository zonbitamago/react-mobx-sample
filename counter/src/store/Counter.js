import { observable } from "mobx";

const Counter = observable.object({
  counter: 0,
  get getCount() {
    return this.counter;
  }
});

export default Counter;
