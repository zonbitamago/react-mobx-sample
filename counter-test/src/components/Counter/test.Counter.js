import React from "react";
import { shallow } from "enzyme";

import Counter from "./Counter";

describe("Counter", () => {
  let component, props;

  beforeEach(() => {
    props = { store: { counter: 0 } };
    component = shallow(<Counter {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});
