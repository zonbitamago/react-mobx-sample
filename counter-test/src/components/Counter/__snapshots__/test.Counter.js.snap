// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    store={
      Object {
        "counter": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Counter:
          0
        </p>,
        <button
          id="button"
          onClick={undefined}
        >
          Add
        </button>,
        <button
          id="minus"
          onClick={undefined}
        >
          minus
        </button>,
      ],
      "className": "Counter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Counter:",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Counter:",
          0,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add",
          "id": "button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Add",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "minus",
          "id": "minus",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "minus",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Counter:
            0
          </p>,
          <button
            id="button"
            onClick={undefined}
          >
            Add
          </button>,
          <button
            id="minus"
            onClick={undefined}
          >
            minus
          </button>,
        ],
        "className": "Counter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Counter:",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Counter:",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "id": "button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "minus",
            "id": "minus",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "minus",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
